autogpt_platform/backend/backend/integrations/creds_manager.py

```diff
@@ -112,26 +111,26 @@ def acquire(
         # to allow priority access for refreshing/updating the tokens.
         with self._locked(user_id, credentials_id, "!time_sensitive"):
             lock = self._acquire_lock(user_id, credentials_id)
-        credentials = self.get(user_id, credentials_id, lock=False)
+        credentials = asyncio.run(self.get(user_id, credentials_id, lock=False))
         if not credentials:
             raise ValueError(
                 f"Credentials #{credentials_id} for user #{user_id} not found"
             )
         return credentials, lock
 
-    def update(self, user_id: str, updated: Credentials) -> None:
+    async def update(self, user_id: str, updated: Credentials) -> None:
         with self._locked(user_id, updated.id):
-            self.store.update_creds(user_id, updated)
+            await self.store.update_creds(user_id, updated)
 
-    def delete(self, user_id: str, credentials_id: str) -> None:
+    async def delete(self, user_id: str, credentials_id: str) -> None:
         with self._locked(user_id, credentials_id):
-            self.store.delete_creds_by_id(user_id, credentials_id)
+            await self.store.delete_creds_by_id(user_id, credentials_id)
 
     # -- Locking utilities -- #
 
     def _acquire_lock(self, user_id: str, credentials_id: str, *args: str) -> RedisLock:
         key = (
-            self.store.supabase.supabase_url,
+            self.store.prisma,
```

Comment:
We might be wanting passing DatabaseManager instead of prisma here.

Reply:
Done